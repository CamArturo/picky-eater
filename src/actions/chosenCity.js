export const addChosenCity = (city) => {
  return {
    type: 'ADD_CHOSEN_CITY',
    city
  }
};