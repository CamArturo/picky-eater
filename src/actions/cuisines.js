export const sendCuisinesToStore = (cuisines) => {
  return {
    type: 'ADD_AVAILABLE_CUISINES',
    cuisines
  };
};