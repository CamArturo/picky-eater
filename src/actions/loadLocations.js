export const sendLocationsToStore = (locations) => (
  {
    type: 'STORE_CITIES',
    locations
  }
);