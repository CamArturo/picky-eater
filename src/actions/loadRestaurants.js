export const sendRestaurantsToStore = (restaurants) => (
  {
    type: 'STORE_RESTAURANTS',
    restaurants
  }
);