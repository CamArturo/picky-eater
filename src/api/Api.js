// import React from 'react';
import key from '../apiKey.js';

export const getCities = async (query) => {
  try {
    const baseURL = `https://developers.zomato.com/api/v2.1/cities?q=${query}`;
    const options = {
      method: 'GET',
      body: JSON.stringify(),
      headers: {
        'user-key': key,
        'Content-Type': 'application/json'
      }
    };
    const response = await fetch(baseURL, options);
    const cities = await response.json();
    // console.log(cities)
    return cities;
  } catch (error) {
    throw new Error(`Failed to fetch. (error: ${error.message})`)
  }
};

// export default connect(null, mapDispatchToProps)(getCities);