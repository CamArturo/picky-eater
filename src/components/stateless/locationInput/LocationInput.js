import React from 'react';

const LocationInput = () => {
  return (
    <div>
      <section className="location-container-header">
        <h2>Where are you?</h2>
      </section>
      <section className="locations">
        <h3>Confirm your city</h3>
        <form className="login-form">
          <input type="text" placeholder="Enter City" />
          <button>Submit</button>
        </form>
      </section>
    </div>
  );
};

export default LocationInput;
