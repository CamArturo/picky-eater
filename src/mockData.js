export const chosenCity = {
  city: 'Newton, IL',
  id: 5172
};

export const mockLocations = [
  {
    'id': 7171,
    'name': 'Newton Grove, NC',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 95,
    'state_name': 'North Carolina',
    'state_code': 'NC'
  },
  {
    'id': 5172,
    'name': 'Newton, IL',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 82,
    'state_name': 'Illinois',
    'state_code': 'IL'
  },
  {
    'id': 6946,
    'name': 'Newton, MS',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 93,
    'state_name': 'Mississippi',
    'state_code': 'MS'
  }
];

export const availableCuisines = [
  {
    'cuisine': {
      'cuisine_id': 6,
      'cuisine_name': 'Afghani'
    }
  },
  {
    'cuisine': {
      'cuisine_id': 152,
      'cuisine_name': 'African'
    }
  },
  {
    'cuisine': {
      'cuisine_id': 1,
      'cuisine_name': 'American'
    }
  }
];

export const mockRestaurants = {
  restaurants:
    {
      results_found: 7,
      results_start: 0,
      results_shown: 7,
      restaurants:
        [[Object],
          [Object],
          [Object],
          [Object],
          [Object],
          [Object],
          [Object]]
    }
};

export const mockListRestaurants = [
  'Yesterday\'s Pub & Dining',
  'Olde Tyme Steakhaus',
  'Brew City Pub & Grill',
  'Holiday Restaurant',
  'Richard\'s Farm Restaurant',
  'Lone Star Steakhouse & Saloon',
  'Gray\'s'
];