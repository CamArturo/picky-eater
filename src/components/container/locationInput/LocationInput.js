import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getCities } from '../../../api/Api';
import { sendLocationsToStore } from '../../../actions/loadLocations';
import { connect } from 'react-redux';

class LocationInput extends Component {
  constructor (props) {
    super(props);
    this.state = {
      userInput: ''
    };
  }

  storeCities = async (userInput) => {
    const cities = await getCities(userInput);
    this.props.loadLocations(cities.location_suggestions);
  };

  render () {
    return (
      <div>
        <section className="location-container-header">
          <h2>Where are you?</h2>
        </section>
        <section className="locations">
          <h3>Confirm your city</h3>
          <form className="login-form" onSubmit={(event) => {
            event.preventDefault();
            this.storeCities(this.state.userInput)
          }}>
            <input
              type="text"
              placeholder="Enter City"
              className="search-city-input"
              onChange={(event) => this.setState({userInput: event.target.value})}
            />
            <button onClick={() => this.storeCities(this.state.userInput)} type="button">Submit</button>
          </form>
        </section>
      </div>
    );
  }
}

LocationInput.propTypes = {
  getCities: PropTypes.func.isRequired,
  loadLocations: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  loadLocations: (cities) => dispatch(sendLocationsToStore(cities))
});

export default connect(null, mapDispatchToProps)(LocationInput);