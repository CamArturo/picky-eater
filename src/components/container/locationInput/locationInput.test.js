import React from 'react';
import { LocationInput } from './LocationInput';
import { shallow } from 'enzyme';
import { getCities } from '../../../api/getCities';
import * as mock from '../../../mockData';
import { mapDispatchToProps } from './LocationInput';
import { sendLocationsToStore } from '../../../actions/loadLocations'
import { addChosenCity } from '../../../actions/chosenCity';


jest.mock('../../../api/getCuisines');
// import { mockLocations } from '../../../mockData';

describe('Locations', () => {
  let wrapper;
  let mockProps;
  let loadLocations;
  let mockState;
  let storeCities;

  beforeEach(() => {
    mockState = {
      userInput: ''
    };

    mockProps = {
      loadLocations: jest.fn()
    };

    // loadLocations = jest.fn();
    wrapper = shallow(
      <LocationInput
        loadLocations={loadLocations}
        getCities={getCities}
        {...mockState}
        {...mockProps}
      />);
  });

  // const spy = spyOn(wrapper.instance(), 'getCities');
  // expect(spy).toHaveBeenCalled();

  it('Should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state(user Input) on user Input', () => {
    const expected = 'Newton';
    const mockEvent = {target: {value: 'Newton'}};
    wrapper.find('.search-city-input').simulate('change', mockEvent);
    expect(wrapper.state('userInput')).toEqual(expected);
  });
  it('should call getCities when button is clicked', () => {

    const storeCities = wrapper.instance().storeCities = jest.fn();
    wrapper.find('button').simulate('click');
    expect(storeCities).toHaveBeenCalled();
  });
  it('should call storeCities on Submit of form', () => {

    const storeCities = wrapper.instance().storeCities = jest.fn();
    const event = {preventDefault: jest.fn()};
    wrapper.find('form').simulate('submit', event);
    expect(storeCities).toHaveBeenCalled();

  });
  it('should fire loadLocations when submit button is clicked', () => {
    const mockedDispatchToProps = jest.fn();
    const mappedProps = mapDispatchToProps(mockedDispatchToProps);

    mappedProps.loadLocations(mock.mockLocations);

    const expected = sendLocationsToStore(mock.mockLocations); // everything that is going to be called with
    expect(mockedDispatchToProps).toHaveBeenCalledWith(expected);

  });
});
