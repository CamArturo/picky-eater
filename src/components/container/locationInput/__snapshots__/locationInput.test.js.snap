// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationInput
    getCities={[Function]}
    loadLocations={[Function]}
    userInput=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="location-container-header"
>
          <h2>
                    Where are you?
          </h2>
</section>,
        <section
          className="locations"
>
          <h3>
                    Confirm your city
          </h3>
          <form
                    className="login-form"
                    onSubmit={[Function]}
          >
                    <input
                              className="search-city-input"
                              onChange={[Function]}
                              placeholder="Enter City"
                              type="text"
                    />
                    <button
                              onClick={[Function]}
                              type="button"
                    >
                              Submit
                    </button>
          </form>
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            Where are you?
</h2>,
          "className": "location-container-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Where are you?",
          },
          "ref": null,
          "rendered": "Where are you?",
          "type": "h2",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Confirm your city
</h3>,
            <form
              className="login-form"
              onSubmit={[Function]}
>
              <input
                            className="search-city-input"
                            onChange={[Function]}
                            placeholder="Enter City"
                            type="text"
              />
              <button
                            onClick={[Function]}
                            type="button"
              >
                            Submit
              </button>
</form>,
          ],
          "className": "locations",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Confirm your city",
            },
            "ref": null,
            "rendered": "Confirm your city",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="search-city-input"
                  onChange={[Function]}
                  placeholder="Enter City"
                  type="text"
/>,
                <button
                  onClick={[Function]}
                  type="button"
>
                  Submit
</button>,
              ],
              "className": "login-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search-city-input",
                  "onChange": [Function],
                  "placeholder": "Enter City",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="location-container-header"
>
            <h2>
                        Where are you?
            </h2>
</section>,
          <section
            className="locations"
>
            <h3>
                        Confirm your city
            </h3>
            <form
                        className="login-form"
                        onSubmit={[Function]}
            >
                        <input
                                    className="search-city-input"
                                    onChange={[Function]}
                                    placeholder="Enter City"
                                    type="text"
                        />
                        <button
                                    onClick={[Function]}
                                    type="button"
                        >
                                    Submit
                        </button>
            </form>
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              Where are you?
</h2>,
            "className": "location-container-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Where are you?",
            },
            "ref": null,
            "rendered": "Where are you?",
            "type": "h2",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Confirm your city
</h3>,
              <form
                className="login-form"
                onSubmit={[Function]}
>
                <input
                                className="search-city-input"
                                onChange={[Function]}
                                placeholder="Enter City"
                                type="text"
                />
                <button
                                onClick={[Function]}
                                type="button"
                >
                                Submit
                </button>
</form>,
            ],
            "className": "locations",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Confirm your city",
              },
              "ref": null,
              "rendered": "Confirm your city",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="search-city-input"
                    onChange={[Function]}
                    placeholder="Enter City"
                    type="text"
/>,
                  <button
                    onClick={[Function]}
                    type="button"
>
                    Submit
</button>,
                ],
                "className": "login-form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "search-city-input",
                    "onChange": [Function],
                    "placeholder": "Enter City",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
